include/boost/beast/core/async_base.hpp

```diff
@@ -186,7 +186,22 @@ class async_base
         "Executor type requirements not met");
 
     Handler h_;
-    net::executor_work_guard<Executor1> wg1_;
+    detail::select_work_guard_t<Executor1> wg1_;
+
+public:
+    /** The type of executor associated with this object.
+
+    If a class derived from @ref async_base is a completion
+    handler, then the associated executor of the derived class will
+    be this type.
+*/
+    using executor_type = typename
+        net::associated_executor<
+            Handler,
+            typename detail::select_work_guard_t<Executor1>::executor_type
```

Comment:
`detail` identifiers should not appear in the documentation, which will happen here